src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java

```diff
@@ -128,14 +136,27 @@ public ResponseEntity<String> createCommons(
       throw new CoffeeException();
     }
 
-    Commons commons = Commons.builder()
-      .name(params.getName())
-      .cowPrice(params.getCowPrice())
-      .milkPrice(params.getMilkPrice())
-      .startingBalance(params.getStartingBalance())
-      .startingDate(params.getStartingDate())
-      .leaderboard(params.getLeaderboard())
-      .build();
+    if (params.getDegradationRate() >= 0){
+      commons = Commons.builder()
+        .name(params.getName())
+        .cowPrice(params.getCowPrice())
+        .milkPrice(params.getMilkPrice())
+        .startingBalance(params.getStartingBalance())
+        .startingDate(params.getStartingDate())
+        .leaderboard(params.getLeaderboard())
+        .degradationRate(params.getDegradationRate())
+        .build();
+    } else {
+      commons = Commons.builder()
+        .name(params.getName())
+        .cowPrice(params.getCowPrice())
+        .milkPrice(params.getMilkPrice())
+        .startingBalance(params.getStartingBalance())
+        .startingDate(params.getStartingDate())
+        .leaderboard(params.getLeaderboard())
+        .degradationRate(0)
+        .build();
+    }
```

Comment:
Oh, no you didn't.

I think you know better than this.

There are least two ways to write this better, more DRY.

Reply:
Here's one:  Just one block of builder code, with this:

```
.leaderboard( params.getDegradationRate() >= 0 ?
   params.getDegradationRate()  : 0 )
```